tests/api2/test_legacy_websocket.py

```diff
@@ -1,23 +1,70 @@
+import errno
+import random
+import string
+import time
+
 import pytest
 
-from truenas_api_client import Client
+from truenas_api_client import Client, ClientException
 
+from middlewared.test.integration.assets.account import unprivileged_user
 from middlewared.test.integration.assets.cloud_sync import credential
 from middlewared.test.integration.utils import password, websocket_url
 
 
 @pytest.fixture(scope="module")
 def c():
     with Client(websocket_url() + "/websocket") as c:
-        c.call("auth.login_ex", {
-            "mechanism": "PASSWORD_PLAIN",
-            "username": "root",
-            "password": password(),
-        })
-        yield c
+        for i in range(1, 21):
+            try:
+                c.call("auth.login_ex", {
+                    "mechanism": "PASSWORD_PLAIN",
+                    "username": "root",
+                    "password": password(),
+                })
+            except ClientException as e:
+                if e.errno == errno.EBUSY:
+                    time.sleep(61)
+                    continue
+                else:
+                    raise
+            else:
+                yield c
+                break
+
 
+@pytest.fixture(scope="module")
+def unprivileged_client():
+    suffix = "".join([random.choice(string.ascii_lowercase + string.digits) for _ in range(8)])
+    with unprivileged_user(
+        username=f"unprivileged_{suffix}",
+        group_name=f"unprivileged_users_{suffix}",
+        privilege_name=f"Unprivileged users ({suffix})",
+        allowlist=[],
+        roles=["READONLY_ADMIN"],
+        web_shell=False,
+    ) as t:
+        with Client(websocket_url() + "/websocket") as c:
+            for i in range(1, 21):
+                try:
+                    c.call("auth.login_ex", {
+                        "mechanism": "PASSWORD_PLAIN",
+                        "username": t.username,
+                        "password": t.password,
+                    })
+                except ClientException as e:
+                    if e.errno == errno.EBUSY:
+                        time.sleep(61)
```

Comment:
No, this is too much time wasted. You can make a call to `rate.limit.cache_clear` so we don't have to sleep here.